src/backend/api/Fusion.Resources.Logic/Requests/Queries/ResolveResponsibleDepartment.cs

```diff
@@ -21,17 +22,21 @@ public ResolveResponsibleDepartment(Guid requestId)
         public class Handler : IRequestHandler<ResolveResponsibleDepartment, string?>
         {
             private readonly ResourcesDbContext resourcesDb;
+            private readonly IProjectOrgResolver orgResolver;
+            private readonly IMediator mediator;
 
-            public Handler(ResourcesDbContext resourcesDb)
+            public Handler(ResourcesDbContext resourcesDb, IProjectOrgResolver orgResolver, IMediator mediator)
             {
                 this.resourcesDb = resourcesDb;
+                this.orgResolver = orgResolver;
+                this.mediator = mediator;
             }
 
             public async Task<string?> Handle(ResolveResponsibleDepartment request, CancellationToken cancellationToken)
             {
                 var dbRequest = await resourcesDb.ResourceAllocationRequests.FirstAsync(r => r.Id == request.RequestId, cancellationToken);
 
-                return await new RequestRouter(resourcesDb).RouteAsync(dbRequest, cancellationToken);
+                return await new RequestRouter(resourcesDb, orgResolver, mediator).RouteAsync(dbRequest, cancellationToken);
```

Comment:
Any reason for having RequestRouter class.
Couldn't it all just be located in this handler, or having it like an injectable IRequestRouter something?

Reply:
Extracted an interface. The logic for routing requests could potentially be usable in more places than just when assigning the request.